LibAddonMenu-2.0/LibAddonMenu-2.0/controls/iconpicker.lua

```diff
@@ -134,9 +134,11 @@ function IconPickerMenu:Initialize(name)
 end
 
 function IconPickerMenu:OnMouseEnter(icon)
-    InitializeTooltip(InformationTooltip, icon, TOPLEFT, 0, 0, BOTTOMRIGHT)
-    SetTooltipText(InformationTooltip, LAM.util.GetStringFromValue(icon.tooltip))
-    InformationTooltipTopLevel:BringWindowToTop()
+    if icon.tooltip and icon.tooltip ~= "" then

```

Comment:
Not sure if this check is enough? GetStringFromValue could return an empty string too. Maybe just store the result string in a local variable and check that instead?

Reply:
GetStringFromValue could return an empty string, but even an empty string is not worth to show an empty tooltip

Reply:
I understand that point, but you are not checking if GetStringFromValue returns an empty string right now, so you'd still show the tooltip in that case.

Reply:
now i've changed it to check GetStringFromValue first